File: задачи/3A2B1C.c
#include <stdio.h>

int main(){

    FILE *f = fopen("file.txt", "r");

    char current = fgetc(f);

    int count = 1;

    while (current != EOF){
        char next = fgetc(f);
        if (current == next) count++;
            else {
                printf("%d%c", count, current);
                count = 1;
                current = next;
            }
    }

    fclose(f);
    return 0;
}

File: задачи/3A2B1C.pas
Program P;

var 
    f: Text;
    current, next: char;
    count: integer;


begin
    Assign(f, 'file.txt');
    Reset(f);

    if not EOF(f) then
    begin
        Read(f, current);
        count := 1;

        while not EOF(f) do
        begin
            Read(f, next);

            if current = next then
            begin
                inc(count);
            end
            else
            begin
                Write(count, current);
                count := 1;
                current := next;
            end;
        end;

        Write(count, current);
    end;

    Close(f);
end.
